Src/FluentAssertions/Collections/StringCollectionAssertions.cs

```diff
@@ -190,5 +191,50 @@ public AndConstraint<StringCollectionAssertions> Contain(IEnumerable<string> exp
             args.AddRange(becauseArgs);
             return base.NotContain(unexpected, because, args.ToArray());
         }
+
+        /// <summary>
+        /// Asserts that the collection contains at least one string that matches a wildcard pattern.
+        /// </summary>
+        /// <param name="wildcardPattern">
+        /// The wildcard pattern with which the subject is matched, where * and ? have special meanings.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<StringCollectionAssertions> ContainMatch(string wildcardPattern, string because = "",
+            params object[] becauseArgs)
+        {
+            if (Subject is null)
```

Comment:
You can use the `ForCondition` instead of the `if`. See https://github.com/fluentassertions/fluentassertions/blob/af2830f66184d6e53be78b3a0e858ac3536904ee/Src/FluentAssertions/Equivalency/GenericDictionaryEquivalencyStep.cs#L188

Reply:
I looked into the example you provided, but I'm not sure I'm understanding. We want the method to return early, I still need the `if` clause.

Reply:
You should be able to write something like this
```c#
Execute.Assertion
    .ForCondition(Subject is object)
    .BecauseOf(because, becauseArgs)
    .FailWith("Expected {context:collection} to contain a match of {0}{reason}, but found <null>.", wildcardPattern);
```

If the condition in `ForCondition` fails, then `FailWith` throws a test framework exception (for the FA tests an `XunitException`)

Reply:
@jnyrup That I understood, but if there is no early return there, the method execution continues, and the tests fail.

Reply:
What test fails? 
I just tried to:
* Check out this branch
* Replace the `if` with inner `return` to the code above
* All tests succeeded.

Reply:
I'm sorry, can you post the full snippet. The only way I can think of is saving the `isMatch` to a method scoped variable and adding another `ForCondition` in the final `Assertion`.

Reply:
You can replace
```c#
if (Subject is null)
{
	Execute.Assertion
		.BecauseOf(because, becauseArgs)
		.FailWith("Expected {context:collection} to contain a match of {0}{reason}, but found <null>.", wildcardPattern);

	return new AndConstraint<StringCollectionAssertions>(this);
}
```
with
```c#
Execute.Assertion
    .ForCondition(Subject is object)
    .BecauseOf(because, becauseArgs)
    .FailWith("Expected {context:collection} to contain a match of {0}{reason}, but found <null>.", wildcardPattern);
```

The part below using `isMatch` looks great 👍

Reply:
Oh, I saw my mistake. I had put `.ForCondition(Subject is null)` because that was the `if` condition. Got a bit mislead there, `ForCondition` takes the opposite...

Thanks for the help.